problem/views.py

```diff
@@ -69,6 +70,10 @@ def detail(request, pid):
     tag_form = TagForm()
     try:
         problem = Problem.objects.get(pk=pid)
```

Comment:
If the problem is invisible and request user is now owenr or admin, raise permission deny